iac-extensions/kubernetes/src/test/java/org/sonar/iac/kubernetes/checks/KubernetesVerifier.java

```diff
@@ -72,38 +76,52 @@ public class KubernetesVerifier {
+    var inputFileContext = initialization.first();
+    var commentsVisitor = initialization.second();
+    var projectContext = prepareProjectContext(inputFileContext, fileNames);
+    Verifier.verify(PARSER, inputFileContext, check, multiFileVerifier -> new KubernetesTestContext(multiFileVerifier, inputFileContext, projectContext), commentsVisitor);
   }
 
-  public static void verify(String templateFileName, IacCheck check, Verifier.Issue... expectedIssues) {
-    if (containsHelmContent(templateFileName)) {
-      HelmVerifier.verify(templateFileName, check, expectedIssues);
-    } else {
-      Verifier.verify(YAML_PARSER, BASE_DIR.resolve(templateFileName), check, expectedIssues);
-    }
+  public static void verify(String templateFileName, IacCheck check, Collection<Verifier.Issue> expectedIssues) {
+    var initialization = initializeVerification(templateFileName);
+    var inputFileContext = initialization.first();
+    var commentsVisitor = initialization.second();
+    var projectContext = prepareProjectContext(inputFileContext);
+    Verifier.verify(PARSER, inputFileContext, check, multiFileVerifier -> new KubernetesTestContext(multiFileVerifier, inputFileContext, projectContext), commentsVisitor,
+      expectedIssues.stream().toList());
   }
 
   /**
    * Only usable for pure Kubernetes files
    */
   public static void verifyContent(String content, IacCheck check) {
-    Verifier.verify(YAML_PARSER, content, check);
+    Verifier.verify(PARSER, content, check);
   }
 
-  public static void verifyNoIssue(String templateFileName, IacCheck check) {
+  public static void verifyNoIssue(String templateFileName, IacCheck check, String... fileNames) {
+    var initialization = initializeVerification(templateFileName, fileNames);
+    var inputFileContext = initialization.first();
+    var commentsVisitor = initialization.second();
+    var projectContext = prepareProjectContext(inputFileContext, fileNames);
+    Verifier.verifyNoIssue(PARSER, inputFileContext, check, multiFileVerifier -> new KubernetesTestContext(multiFileVerifier, inputFileContext, projectContext), commentsVisitor);
+  }
+
+  private static Tuple<HelmInputFileContext, BiConsumer<Tree, Map<Integer, Set<Comment>>>> initializeVerification(String templateFileName, String... fileNames) {
     if (containsHelmContent(templateFileName)) {
-      HelmVerifier.verifyNoIssue(templateFileName, check);
+      if (fileNames.length > 0) {
+        throw new IllegalArgumentException("For Helm projects, all project files will be discovered. Explicit input is not required.");
+      }
+      var inputFileContext = HelmVerifier.prepareHelmContext(templateFileName);
+      return new Tuple<>(inputFileContext, HelmVerifier.commentsWithShiftedTextRangeVisitor(inputFileContext));
     } else {
-      Verifier.verifyNoIssue(YAML_PARSER, BASE_DIR.resolve(templateFileName), check);
+      var inputFileContext = new HelmInputFileContext(SENSOR_CONTEXT, inputFile(templateFileName, BASE_DIR));
```

Comment:
If possible use already normal `InputFileContext` for k8s files. Otherwise I will change it later in my PR

Reply:
For now it's needed for the `prepareProjectContext` method, so I'll leave it like this for now